### DISCUSSION TITLE
Order matters?

### DISCUSSION THREAD

**Discussion by lawrencegrass** [victools/jsonschema-generator#392] [Q&A] [answered]
I'm trying to get the config to not have the generator re-order the properties when creating the schema.  In other words, maintain the order the fields are declared in the POJO.

Any guidance would be awesome.

**Comment by CarstenWickner** [comment, marked as answer]
Hi @lawrencegrass,

Fortunately, this had been raised before in #92, i.e., yes: this is possible.

If your compiler happens to honour the declaration order, then you can disable the property sorting with this:
```java
configBuilder.forTypesInGeneral()
        .withPropertySorter((_prop1, _prop2) -> 0);
```

The standard sorting logic is defined in the `PropertySortUtils` class, in case you want to apply some other sorting behaviour.

Additionally, the `JacksonModule` supports to sort an object's properties according to its `@JsonPropertyOrder` annotation, if the `JacksonOption.RESPECT_JSONPROPERTY_ORDER` was provided
